pipeline/rust/rust.go

```diff
@@ -0,0 +1,213 @@
+package rust
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/goreleaser/goreleaser/config"
+	"github.com/goreleaser/goreleaser/context"
+	"github.com/goreleaser/goreleaser/internal/artifact"
+	"github.com/goreleaser/goreleaser/pipeline"
+)
+
+// Pipe for rust
+type Pipe struct{}
+
+func (Pipe) String() string {
+	return "building rust binaries"
+}
+
+// Run the pipe
+func (Pipe) Run(ctx *context.Context) error {
+	if len(ctx.Config.Rust) == 0 {
+		return pipeline.Skip("rust section is not configured")
+	}
+
+	for _, rust := range ctx.Config.Rust {
+		if len(rust.Target) == 0 {
+			return pipeline.Skip("rust section is not configured properly (missing targets)")
+		}
+
+		log.WithField("rust", rust).Debug("building")
+		if err := runPipeOnBuild(ctx, rust); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Default sets the pipe defaults
+func (Pipe) Default(ctx *context.Context) error {
+	// only set defaults if there are rusts in the config file.
+	if len(ctx.Config.Rust) == 0 {
+		return nil
+	}
+
+	for i, rust := range ctx.Config.Rust {
+		ctx.Config.Rust[i] = buildWithDefaults(ctx, rust)
+	}
+	return nil
+}
+
+func buildWithDefaults(ctx *context.Context, rust config.Rust) config.Rust {
+	if rust.Binary == "" {
+		rust.Binary = ctx.Config.Release.GitHub.Name
+	}
+
+	// One idea would be, if no rust.Target are defined
+	// to add the default target for the current OS.
+	// E.g. hit `rustup show`. The default host is shown.
+	// For an up to date Mac is will show "Default host: x86_64-apple-darwin"
```

Comment:
I think we could default this to the same targets as go:

- x86_64-apple-darwin
- x86_64-unknown-linux-gnu
- i686-apple-darwin
- i686-unknown-linux-gnu

Reply:
We thought about this as well. Like If MacOSx than target this. If linux, then this and so on. But we have to take the architecture and even the linker-system into account (when it comes to GNU libc or Musl libc). And there it is getting hard to decide what is the default

Reply:
Won't be easy. See comment [here](https://github.com/goreleaser/goreleaser/pull/520#issuecomment-359570387).